whelk-core/src/test/groovy/whelk/filter/LanguageLinkerSpec.groovy

```diff
@@ -264,16 +280,80 @@ class LanguageLinkerSpec extends Specification {
-        [language: [[label: 'grekiska'], ['@id': 'http://id/gre']]]         | true   | [language: [['@id': 'http://id/gre']]]
-        [language: [[label: 'grekiska'], ['@id': 'http://id/grc']]]         | true   | [language: [['@id': 'http://id/grc']]]
+        data                                                        | change | expected
+        [language: [[label: 'tonga']]]                              | false  | [language: [[label: 'tonga']]]
+        [language: [[label: 'tonga'], ['@id': 'http://id/ton']]]    | true   | [language: [['@id': 'http://id/ton']]]
+        [language: [[label: 'tonga'], ['@id': 'http://id/tog']]]    | true   | [language: [['@id': 'http://id/tog']]]
+
+        [language: [[label: 'grekiska']]]                           | false  | [language: [[label: 'grekiska']]]
+        [language: [[label: 'grekiska'], ['@id': 'http://id/gre']]] | true   | [language: [['@id': 'http://id/gre']]]
+        [language: [[label: 'grekiska'], ['@id': 'http://id/grc']]] | true   | [language: [['@id': 'http://id/grc']]]
     }
 
+    def "handles ambiguous labels using given input to disambiguate"() {
+        given:
+        LanguageLinker linker = new LanguageLinker()
+        [
+                [
+                        '@id'          : 'http://id/tog',
+                        code           : 'tog',
+                        prefLabelByLang: [
+                                'sv': 'Tonga'
+                        ],
+                        commentByLang  : [
+                                sv: 'Nyasa'
+                        ]
+                ],
+                [
+                        '@id'          : 'http://id/ton',
+                        code           : 'ton',
+                        prefLabelByLang: [
+                                sv: 'Tonga'
+                        ],
+                        commentByLang  : [
+                                sv: 'Tongaöarna'
+                        ]
+                ],
+                [
+                        '@id'          : 'http://id/gre',
+                        code           : 'gre',
+                        prefLabelByLang: [
+                                'sv': 'Nygrekiska'
+                        ]
+                ],
+                [
+                        '@id'          : 'http://id/grc',
+                        code           : 'grc',
+                        prefLabelByLang: [
+                                'sv': 'Grekiska, klassisk'
+                        ]
+                ],
+                [
+                        '@id'          : 'http://id/lat',
+                        code           : 'lat',
+                        prefLabelByLang: [
+                                'sv': 'Latin',
+                                'en': 'Latin'
+                        ]
+                ]
+        ].each(linker.&addDefinition)
+        linker.addMapping('grekiska', 'http://id/gre')
+        linker.addMapping('grekiska', 'http://id/grc')
```

Comment:
Put these as `hiddenLabelByLang` in the test data?

Reply:
:+1: See [eafb332](https://github.com/libris/librisxl/pull/1174/commits/eafb332a4f8429ad189c9935ec10220c3fc00cf5)